packages/contracts/contracts/default-apps/ETHBucket.sol

```diff
@@ -1,66 +1,31 @@
 pragma solidity 0.5.8;
 pragma experimental "ABIEncoderV2";
 
-import "../libs/Transfer.sol";
-import "../CounterfactualApp.sol";
+import "../interfaces/CounterfactualApp.sol";
+import "../interfaces/Interpreter.sol";
+import "../interpreters/ETHInterpreter.sol";
 
 
-contract ETHBucket is CounterfactualApp {
+contract ETHBucket {
 
   struct AppState {
-    address alice;
-    address bob;
-    uint256 aliceBalance;
-    uint256 bobBalance;
+    ETHInterpreter.ETHTransfer[] transfers;
```

Comment:
Seems clever but makes the data structure unnecessarily nested I think? To be specific, this is now `tuple(tuple(address[], uint256[])[]))` instead of `tuple(address[], uint256[])` right?

Reply:
It should be `tuple(address, uint256)[]`

Reply:
That is not what this is right now as far as I can tell. It is `tuple(tuple(address[], uint256[])[]))`.

Reply:
`ETHTransfer` is a static type with 2 fields. Anyway, I have removed the wrapper functionality and disabled linting for this file.